depends/common/imagemagick/0001-prevent-C-system-calls-for-Kodi.patch

```diff
@@ -0,0 +1,168 @@
+diff --git a/ImageMagick6-6.9.10-25/magick/delegate.c b/ImageMagick6-6.9.10-25/magick/delegate.c
+index 6d9b810..2d88cb7 100644
+--- a/magick/delegate.c
++++ b/magick/delegate.c
+@@ -376,83 +376,83 @@ MagickExport int ExternalDelegateCommand(const MagickBooleanType asynchronous,
+     (void) ConcatenateMagickString(sanitize_command,"&",MaxTextExtent);
+   if (message != (char *) NULL)
+     *message='\0';
+-#if defined(MAGICKCORE_POSIX_SUPPORT)
+-#if !defined(MAGICKCORE_HAVE_EXECVP)
+-  status=system(sanitize_command);
+-#else
+-  if ((asynchronous != MagickFalse) ||
+-      (strpbrk(sanitize_command,"&;<>|") != (char *) NULL))
+-    status=system(sanitize_command);
+-  else
+-    {
+-      pid_t
+-        child_pid;
+-
+-      /*
+-        Call application directly rather than from a shell.
+-      */
+-      child_pid=(pid_t) fork();
+-      if (child_pid == (pid_t) -1)
+-        status=system(sanitize_command);
+-      else
+-        if (child_pid == 0)
+-          {
+-            status=execvp(arguments[1],arguments+1);
+-            _exit(1);
+-          }
+-        else
+-          {
+-            int
+-              child_status;
+-
+-            pid_t
+-              pid;
+-
+-            child_status=0;
+-            pid=(pid_t) waitpid(child_pid,&child_status,0);
+-            if (pid == -1)
+-              status=(-1);
+-            else
+-              {
+-                if (WIFEXITED(child_status) != 0)
+-                  status=WEXITSTATUS(child_status);
+-                else
+-                  if (WIFSIGNALED(child_status))
+-                    status=(-1);
+-              }
+-          }
+-    }
+-#endif
+-#elif defined(MAGICKCORE_WINDOWS_SUPPORT)
+-  {
+-    register char
+-      *p;
+-
+-    /*
+-      If a command shell is executed we need to change the forward slashes in
+-      files to a backslash. We need to do this to keep Windows happy when we
+-      want to 'move' a file.
+-
+-      TODO: This won't work if one of the delegate parameters has a forward
+-            slash as aparameter.
+-    */
+-    p=strstr(sanitize_command, "cmd.exe /c");
+-    if (p != (char*) NULL)
+-      {
+-        p+=10;
+-        for (; *p != '\0'; p++)
+-          if (*p == '/')
+-            *p=*DirectorySeparator;
+-      }
+-  }
+-  status=NTSystemCommand(sanitize_command,message);
+-#elif defined(macintosh)
+-  status=MACSystemCommand(sanitize_command);
+-#elif defined(vms)
+-  status=system(sanitize_command);
+-#else
+-#  error No suitable system() method.
+-#endif
++// #if defined(MAGICKCORE_POSIX_SUPPORT)
```

Comment:
IMO commenting out the lines instead of removing them is of no use.

Reply:
About the remove. I more want to leave it commented.
Here and also on the most addons where this patch ways used are some Hacks done to become it working for Kodi.

If the code is removed and there comes a fault is it hard to see during debug that there was something before.